### DISCUSSION TITLE
tutorials/navigation/navigation_using_navigationregions

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#254] [User-contributed notes]
# tutorials/navigation/navigation_using_navigationregions

NavigationRegions are the visual Node representation of a region of the navigation map on the NavigationServer. Each NavigationRegion node holds a resource for the navigation mesh data. Both 2D and...

https://docs.godotengine.org/en/stable/tutorials/navigation/navigation_using_navigationregions.html

**Comment by boholstapp** [comment]
video Example
https://youtu.be/qIqcp7xBGkw

**Reply by gamis-cool** [reply]
thank you king

**Comment by MarkoGrbec** [comment]
if there is CSG subtracted and you bake at run time to correctly have it baked you need to wait 1 frame when the CSG meshes were deployed or make next code if it's at start of NavigationRegion3D

	extends NavigationRegion3D
	
	func _ready() -> void:
		await get_tree().process_frame
		bake_navigation_mesh()
